### ISSUE TITLE
UnnecessaryParentheses false positive when using `?:` and `>`

### ISSUE THREAD

**Issue by PavlosTze** [detekt/detekt#6889]
## Expected Behavior
`((x ?: 0.0F) > 0)` should not produce an error.

## Observed Behavior
This `((x ?: 0.0F) > 0)` produces an `UnnecessaryParentheses` with the hint that it can be used as `(x ?: 0.0F > 0)` but Android Studio shows a warning that clarifying parentheses should be used.
![εικόνα](https://github.com/detekt/detekt/assets/19330987/1d6104bc-839d-4851-964e-5e95caf1a8ed)

## Steps to Reproduce
Enable `UnnecessaryParentheses` and use the following code:
`val y = ((x ?: 0.0F) > 0)`

## Context
My goal was to optimize my detekt rules by ended up not using `UnnecessaryParenetheses` because of that.

## Your Environment
* Version of detekt used: 1.23.4
* Operating System and version: Windows 11
* Link to your project (if it's a public repository): https://github.com/WeatherXM/wxm-android
* Android Studio: Android Studio Hedgehog | 2023.1.1 Patch 1 - Build #AI-231.9392.1.2311.11255304, built on December 27, 2023

**Comment by marschwar** [comment]
You could set `allowForUnclearPrecedence` to true. See https://detekt.dev/docs/rules/style/#unnecessaryparentheses

```kotlin
// allowForUnclearPrecedence=false => 2 violations
// allowForUnclearPrecedence=true => 1 violation
val y = ((x ?: 0.0F) > 0)

// allowForUnclearPrecedence=false => 1 violation
// allowForUnclearPrecedence=true => ok
val y = (x ?: 0.0F) > 0

```

**Comment by PavlosTze** [comment]
You are correct, it fixed my issue! Thank you!
